frontend/src/components/Navigator/index.js

```diff
@@ -158,6 +164,24 @@ function Navigator({
     };
   });
 
+  const { socket, store, clearStore } = useContext(SocketContext);
+  const { user } = useContext(UserContext);
+
+  const INTERVAL_DELAY = 4000;
```

Comment:
I think we can remove this heartbeat, we should only need to send an update on which page a user is looking at on navigation (unless I'm missing something). Also I would _think_ the server side gets hit with a websocket "disconnected" no matter how the user leaves the page so everything should get unsubscribed when a user closes their browser (or their computer suddenly shuts off, etc..)

Reply:
We talked on a call about this a bit. I didn't think through the need of having a heartbeat completely. Using the redis pub/sub functionality messages are gone once broadcast, so there is no way for a user that joins a second after a message to receive that message. We might be able to use the autosave heartbeat (every 2 minutes). We will check at standup if a 2 minute lag time is alright or if we need a tighter window